### PR TITLE
Lo L1A CDF YAML Reorganization

### LINKED ISSUE #843: Improve Lo L1A CDF YAML Definitions
### Summary

The Lo L1A CDF YAML definitions should be re-worked in the following ways
- make better use of shared attributes
- change to use int32 for fields currently using int64
- add datatype field to specify the data type to be used when creating the data array

### Requirements

_No response_

### Related tickets

_No response_

### PR DESCRIPTION
**Pull request by sdhoyt**
# Change Summary

Reorganized the Lo L1A CDF YAML to make better use of default values and added data type to use in data arrays to match what Hi is currently doing

Closes #843 

# Testing
- Ran existing unit tests
- Ran cli.py to create L1A histogram CDF and spot checked contents

### CODE REVIEW

**Review by maxinelasp — APPROVED** [review]
Looks good, one minor thing that might not even be wrong. Did you double check that you can stack defaults like that and get the expected result?

File: imap_processing/cdf/config/imap_lo_l1a_variable_attrs.yaml [hunk 1]
```diff
@@ -1,133 +1,182 @@
...
+
+default_uint32_attrs: &default_uint32
+  <<: *default
+  FILLVAL: 4294967295
+  FORMAT: I10
+  VALIDMIN: 0
+  VALIDMAX: 4294967295
+  dtype: uint32
+
+default_uint64_attrs: &default_uint64
```
Comment (tmplummer) [inline imap_processing/cdf/config/imap_lo_l1a_variable_attrs.yaml]: I found out the hard way that CDF doesn't have a CDF_UINT8 datatype, so I think that we need to use `int64` only, not `uint64`.

File: imap_processing/cdf/config/imap_lo_l1a_variable_attrs.yaml [hunk 2]
```diff
@@ -1,133 +1,182 @@
...
+
+spin_default_attrs: &spin_default
+  <<: *default
+  VAR_TYPE: support_data
+  DEPEND_1: spin
+
 
 # Re-used Coordinate
+epoch:
+  <<: *default_uint32
```
Comment (tmplummer) [inline imap_processing/cdf/config/imap_lo_l1a_variable_attrs.yaml]: Everything I have seen indicates that we want to use `int64` for epoch.

File: imap_processing/cdf/config/imap_lo_l1a_variable_attrs.yaml [hunk 3]
```diff
@@ -1,133 +1,182 @@
...
+
 esa_step_label:
    CATDESC: ESA Steps
    FIELDNAM: ESA Steps
    FORMAT: A1
    VAR_TYPE: metadata
 
 esa_step:
+  <<: *coord_default
+  <<: *default_uint8
```
Comment (tmplummer) [inline imap_processing/cdf/config/imap_lo_l1a_variable_attrs.yaml:93]: Oh, interesting! I hadn't thought about multiple inheritance like this. Good pattern to keep in mind.

**Review by tmplummer — COMMENTED** [review]
I made two comments worth looking at before merging. Thanks for keeping this to a single file change rather than making the code changes in the same PR!

**Comment by sdhoyt** [comment]
> Looks good, one minor thing that might not even be wrong. Did you double check that you can stack defaults like that and get the expected result?

I spot checked one of the stacked fields in the CDF and it has all the attribute keys with the values I expected, so I think it works as you'd expect.

**Review by tmplummer — APPROVED** [review]

**Pull request merged**